### PR TITLE
fix: Highlights on Fedora.

### LINKED ISSUE #223: Cannot load /usr/lib64/atom/node_modules/highlights
## Description

asciidoc-preview fails to start with a cannot load the /usr/lib64/atom/node_modules/highlights module error.

- Atom version: 1.14.1
- OS: Fedora
- `asciidoc-preview` version: 2.9.0
- `language-asciidoc` version: 1.8.0

Creating the following symlink fixes the issue:

    ln -s /usr/lib64/atom/node_modules/highlight.js/ /usr/lib64/atom/node_modules/highlights

Apparently, asciidoc-preview should point to ``highlight.js`` instead of ``highlights``.

### PR DESCRIPTION
**Pull request by ldez**
## Description

Fix Highlights module path on Fedora.

Fixes #223, #226

### CODE REVIEW

File: lib/renderer.coffee [hunk 1]
```diff
@@ -3,8 +3,16 @@
...
+
 # No direct dependence with Highlight because it requires a compilation. See #63 and #150 and atom/highlights#36.
-Highlights = require path.join atom.packages.resolvePackagePath('markdown-preview'), '..', 'highlights'
+markdownPreviewPath = atom.packages.resolvePackagePath 'markdown-preview'
+commonHighlightsPath = path.join markdownPreviewPath, '..', 'highlights'
+if fs.isDirectorySync commonHighlightsPath
+  Highlights = require commonHighlightsPath
+else
+  # Fix Fedora specific problem #226
+  Highlights = require path.join markdownPreviewPath, 'node_modules', 'highlights'
```
Comment (Mogztter) [inline lib/renderer.coffee]: Is it really related to Fedora ? It looks more like a npm/Node compatibility issue ?

Comment (Mogztter) [inline lib/renderer.coffee]: Or maybe it's just the way Fedora is bundling Atom 😐

Comment (ldez) [inline lib/renderer.coffee]: It seems Atom on Fedora don't use `asar` system.
I think it's a behavior of Electron on Fedora.

Comment (mojavelinux) [inline lib/renderer.coffee]: I just did a fresh install of Atom 1.15.0 (atom.x86_64.rpm) on Fedora 21 (yes, I realize I'm using a very old version of Fedora). The latest AsciiDoc Preview release finds the markdown-preview package without a problem. I can browse in the JavaScript debugger and see the markdown-preview package under /usr/share/atom/resources (which appears to be some sort of virtual file system). If this is a problem, it doesn't affect Fedora 21.

However, while testing, what I discovered is that the colors are no longer applied. It probably has something to do with this message:

> Starting from Atom v1.13.0, the contents of atom-text-editor elements are no longer encapsulated within a shadow DOM boundary. This means you should stop using :host and ::shadow pseudo-selectors, and prepend all your syntax selectors with syntax--.

Comment (mojavelinux) [inline lib/renderer.coffee]: @Mogztter Fedora doesn't bundle Atom. These packages are from the Atom project (unless there is someone else packaging Atom that I don't know of).

**Comment by mojavelinux** [comment]
It appears this workaround is required for the package to work with the RPM provided by the Fedora project (currently in copr). The comment should be updated to reflect that it is not specific to Fedora the OS, but rather the RPM made for Fedora by the Fedora community.

**Pull request merged**